### ISSUE TITLE
Error (or warn) for `default_package` entries that don't have a corresponding `dependencies` entry

### ISSUE THREAD

**Issue by stuartmorgan** [dart-lang/sdk#47682]
I'm not sure if there's any precedent for having errors/warnings that are based on Flutter-specific keys, but I've encountered a case where it would be very helpful.

Flutter's federated plugin structure has a key in the `flutter:plugin:` section that allows indicating that a given platform's implementation is in another package. E.g.:
```
flutter:
  plugin:
    platforms:
      android:
        [...]
      ios:
        [...]
      macos:
        default_package: path_provider_macos
```

This is informational; it's used by `flutter` during the build to determine how to handle conflicts if multiple implementations of the same package are in the dependency tree, and also by pub.dev to determine which platforms to show support for. What actually makes that platform work is including the implementing package (in the example, `path_provider_macos`) as a dependency. The intend of the `default_package` entry is that it's done in conjunction with doing that, in the same pubspec.yaml.

https://github.com/ryanheise/just_audio/issues/582 is an example of a package that was uploaded with a package listed in the `default_package` section, but not in `dependencies`; the result is a package that claims to support a platform which won't actually work. That combination isn't meaningful, so we shouldn't allow it.

(If this isn't something we want to put in dev, we could instead change pub.dev to validate this before showing platform support, but preventing upload of a bad state would be preferable IMO.)

**Comment by bwilkerson** [comment]
Yes, given that there aren't any false positives and it's not a matter of personal preference, we should create a hint for this.

**Comment by goderbauer** [comment]
I am also deferring to @stuartmorgan on this one. The suggestion sounds reasonable to me, though.
